packages/date-picker/src/DatePicker/components/DateInput/DateInput.scss

```diff
@@ -0,0 +1,96 @@
+@import "~@kaizen/design-tokens/sass/color";
+@import "~@kaizen/component-library/styles/animation";
+@import "~@kaizen/deprecated-component-library-helpers/styles/layout";
+@import "~@kaizen/deprecated-component-library-helpers/styles/type";
+@import "~@kaizen/deprecated-component-library-helpers/styles/color";
+@import "~@kaizen/component-library/styles/utils";
+@import "~@kaizen/component-library/styles/mixins/forms";
+@import "../../mixins";
+
+// -----------------------------------------------
+// Date Input
+// -----------------------------------------------
+
+///////////////////////////////////////////////////
+// MIXINS
+///////////////////////////////////////////////////
+
+@mixin input-icon($color, $reversed) {
+  color: $color;
+
+  .withReversed & {
+    color: $reversed;
+  }
+}
+
+// Vars
+$input-disabled-opacity: 0.3;
+
+.iconButton {
+  @include button-reset;
+  display: inherit;
+  padding: 0.25rem $spacing-xs;
+  border: $border-solid-border-width $border-solid-border-style transparent;
+  border-radius: $border-solid-border-radius;
+  position: relative;
+
+  &:hover:not([disabled]) {
+    background-color: $color-blue-100;
+    color: $color-blue-500;
+  }
+
+  &:global(.focus-visible) {
+    outline-offset: -2px;
+    outline: $color-blue-500 $border-focus-ring-border-style
+      $border-focus-ring-border-width;
+  }
+
+  :focus {
+    outline: none;
+  }
+
+  .disabled {
+    opacity: $input-disabled-opacity;
+  }
+
+  &.calendarActive {
+    color: $color-blue-500;
+    background-color: $color-blue-100;
+  }
+
+  svg {
+    margin-top: 2px;
+  }
+}
+
+///////////////////////////////////////////////////
+// VALIDATION STYLES
+///////////////////////////////////////////////////
+
+.success {
+  @include input-icon($color-green-500, $color-green-400);
+  @include ca-animation-fade(in);
+
+  &.disabled {
+    @include input-icon(
+      rgba($color-green-500-rgb, $input-disabled-opacity),
+      rgba($color-green-400-rgb, $input-disabled-opacity)
+    );
+  }
+}
+
+.error {
+  @include input-icon($color-red-100, $color-purple-700);
+  @include ca-animation-fade(in);
+}
+
+.caution {
+  @include input-icon($color-yellow-200, $color-purple-700);
+  @include ca-animation-fade(in);
+}
+
+.message {
+  &.disabled {
+    opacity: $input-disabled-opacity;
+  }
+}
```

Comment:
I think this is more DatePicker implementation but I realized that rtl changes the direction of the calendar month arrows, which should ideally remain in the same direction.

![Screen Shot 2022-04-22 at 12 31 25 pm](https://user-images.githubusercontent.com/36558508/164584883-b1d81397-42bb-42e9-a692-3bc83e3f6396.png)

Reply:
We aren't handling RTL in this iteration :)